GOOD-I-DEER/nodes/cosine-similarity-node/calculate-cosine.js

```diff
@@ -70,10 +72,16 @@ module.exports = function (RED) {
         } else {
           result = devideSimilarity(input_vectors, stored_vectors);
         }
-  
-        msg.payload = result;
-        msg.topid = "Result";
-        this.send(msg);  
+
+        if (!result) {
+          msg.payload = "The Vector is not available to calculate.";
+          msg.topic = "Error";
+          this.send(msg);
```

Comment:
This code may cause unexpected type errors in the node responsible for output.
Don't use send method, just use node.error method and finish the node.

Reply:
it would be better use 'node.error' in error case

Reply:
Checked! Now this handle every error by "this.error()". Please check new commit below 😄